### ISSUE TITLE
Event Subscribers: Can't subscribe the same method to two events

### ISSUE THREAD

**Issue by mamhoff** [solidusio/solidus#4340]
I have an event subscriber that I would like to subscribe to two events: `order_finalized` and `order_recalculated`. 

Based on the [documentation](https://guides.solidus.io/developers/events/overview.html), I set out to create a subscriber like so:

```ruby
module MySubscriber
  include Spree::Event::Subscriber

  event_action :my_method, event_name: :order_recalculated
  event_action :my_method, event_name: :order_finalized

  def my_method(event)
    puts "You've recalculated or finalized your order!
  end
end
``` 

However, my new subscriber will only run when the `order_finalized` event is fired. I think this has to do with the `event_actions` hash in the `Spree::Subscriber` module. Its keys are method names, so in the above example the key would be `:my_method`, and the second invocation of `event_action` will overwrite the previously existing contents of `Spree::Subscriber.event_actions[:my_method]`. See [here](https://github.com/solidusio/solidus/blob/master/core/lib/spree/event/subscriber.rb#L33) and [here](https://github.com/solidusio/solidus/blob/master/core/lib/spree/event/subscriber.rb#L62).

**Solidus Version**:
Reproducible on 2.11 and master branch.

**To Reproduce**
See above.

**Current behavior**

`MySubscriber#my_method` is only called upon firing `order_finalized`, because that is the last invocation of `event_action`.  

**Expected behavior**

`MySubscriber#my_method` is called upon firing `order_recalculated` and `order_finalized`.

**Comment by waiting-for-dev** [comment]
I'll propose replacing the current event system with [Omnes](https://github.com/nebulab/omnes) (which is an extraction of the former, with better subscriber objects) very soon. That is fixed there.

**Comment by waiting-for-dev** [comment]
> I'll propose replacing the current event system with [Omnes](https://github.com/nebulab/omnes) (which is an extraction of the former, with better subscriber objects) very soon.

Please, see https://github.com/solidusio/solidus/pull/4342

**Comment by waiting-for-dev** [comment]
Closed by #4342
